### PR TITLE
Fix recently-introduced faulty slice index comparison

### LINKED ISSUE #3202: New aggregate_slicewise.py check causes error in batch_processing.sh
### Description

[One of the commands](https://github.com/neuropoly/spinalcordtoolbox/pull/3198/checks?check_run_id=1802242428) in `batch_processing.sh` failed the new check added in #3120. 

```python
--
Spinal Cord Toolbox (git-jn/2888-test-batch_processing.sh-881b504ab2ea7140375186c54e2e9b8bb3f5ab51*)

sct_extract_metric -i mtr.nii.gz -method map -o mtr_in_wm.csv -l 51 -vert 2:5
--

ValueError: Shape mismatch between vertfile [22] and metric [61]). Please verify that your vertfile has the same number of slices as your input image.
```

This is my mistake -- the new check is comparing the wrong dimension:

https://github.com/neuropoly/spinalcordtoolbox/blob/49ed68e029886661ad93546f5c0cd981839a6541/spinalcordtoolbox/aggregate_slicewise.py#L237

`[0]` does not refer to the proper slice index for 3D RPI data. Changing to `[-1]` will properly fetch slices from the last dimension of the `Metric()`. 

### Postmortem

This slipped through because tests in `test_aggregate_slicewise.py` use only 1D `Metric()` data.

https://github.com/neuropoly/spinalcordtoolbox/blob/49ed68e029886661ad93546f5c0cd981839a6541/unit_testing/test_aggregate_slicewise.py#L25-L32

Because of this, I falsely assumed all metrics were 1D, however `mtr.nii.gz` is a 3D metric. So, we should include a dummy 3D metric in there as well.

### PR DESCRIPTION
**Pull request by joshuacwnewton**
#### GitHub

- [X] I've given this PR a concise, self-descriptive, and meaningful title
- [X] I've linked relevant issues in the PR body
- [X] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [X] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [SCT's internal developer documentation](https://github.com/neuropoly/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [X] I've added [relevant tests](https://github.com/neuropoly/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/neuropoly/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

In #3120, I mistakenly assumed that all metrics would be 1D, so I indexed the metric using `[0]`. This updates the check to index the right dimension depending on the metric shape.

This PR also adds a new dummy metric that is 3D, so that the tests are more representative of real-world data. 

## Linked issues

Fixes #3202.

### CODE REVIEW

File: spinalcordtoolbox/aggregate_slicewise.py [hunk 1]
```diff
@@ -233,11 +233,12 @@ def aggregate_per_slice_or_level(metric, mask=None, slices=[], levels=[], persli
     :return: Aggregated metric
     """
     if vert_level:
-        vert_level_slices = Image(vert_level).change_orientation('RPI').data.shape[2]
-        metric_slices = metric.data.shape[0]
+        vert_level_slices = Image(vert_level).change_orientation('RPI').data.shape[-1]
+        metric_slices = metric.data.shape[-1]
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/aggregate_slicewise.py:237]: Hrmm, wait a minute. Is `[-1]` what I want? 4D images have `t` as their last dimension, right?

**Comment by joshuacwnewton** [comment]
Note: The build didn't fail from this PR's changes. 

For some reason, the `centos:8` Travis jobs fail intermittently. The job says `Unable to find image 'centos:8' locally`, then it tries to fetch the image, but it only succeeds some of the time. Since the issue happens intermittently, usually restarting the job causes it to pass.

I might open a new issue for this, but considering we'll soon be transitioning away from Travis, I wonder if that's even worth it. For now, I've simply restarted the job.

File: spinalcordtoolbox/aggregate_slicewise.py (hunk 1 above)
Comment (joshuacwnewton) [reply spinalcordtoolbox/aggregate_slicewise.py]: I've changed this to more explicitly index the I-S slices depending on the shape of the data.

**Review by Drulex — APPROVED** [review]
+1 for adding the failed test case!

**Pull request merged**